packages/@uppy/companion/src/server/provider/drive/index.js

```diff
@@ -24,66 +24,80 @@ class Drive {
   list (options, done) {
     const directory = options.directory || 'root'
     const query = options.query || {}
-    const teamDrive = query.teamDrive
-    let listDone = false
-    let teamDrivesDone = false
-    let teamDrives
-    let listResponse
-    let reqErr
-    const finishReq = () => {
-      if (reqErr || listResponse.statusCode !== 200) {
-        const err = this._error(reqErr, listResponse)
-        logger.error(err, 'provider.drive.list.error')
-        done(err)
-      } else {
-        done(null, this.adaptData(listResponse.body, teamDrives ? teamDrives.body : null, options.uppy))
-      }
+    const teamDriveId = query.teamDriveId
+
+    let teamDrivesPromise = Promise.resolve(undefined)
+
+    const shouldListTeamDrives = directory === 'root' && !teamDriveId && !query.nextPageToken
+    if (shouldListTeamDrives) {
+      teamDrivesPromise = new Promise((resolve) => {
+        this.client
+          .query()
+          .get('teamdrives')
+          .where({ fields: TEAM_DRIVE_FIELDS })
+          .auth(options.token)
+          .request((err, resp) => {
+            if (err) {
+              logger.error(err, 'provider.drive.teamDrive.error')
+              return
+            }
+            resolve(resp)
+          })
+      })
+    }
+
+    let where = {
+      fields: DRIVE_FILES_FIELDS,
+      pageToken: query.nextPageToken,
```

Comment:
maybe we should make the nextPage query name the same for all providers? so each provider would use it as it requires internally. But between companion and uppy client it will be the same query name. For instagram we are already using `max_id` but we can change it and still support `max_id` for backward compatibility.